src/test/base32_tests.cpp

```diff
@@ -22,8 +22,8 @@ BOOST_AUTO_TEST_CASE(base32_testvectors)
         BOOST_CHECK_EQUAL(strEnc, vstrOut[i]);
         strEnc = EncodeBase32(vstrIn[i], false);
         BOOST_CHECK_EQUAL(strEnc, vstrOutNoPadding[i]);
-        std::string strDec = DecodeBase32(vstrOut[i]);
-        BOOST_CHECK_EQUAL(strDec, vstrIn[i]);
+        auto dec = DecodeBase32(vstrOut[i]);
+        BOOST_CHECK_MESSAGE(MakeByteSpan(dec) == MakeByteSpan(vstrIn[i]), vstrIn[i]);
```

Comment:
edit: Looks like the binary data in this test are printable-ascii-only strings, so feel free to ignore. Also it looks like boost doesn't attempt to sanitize the binary data anyway.

Though, I still slightly prefer to print both the decoded string and the input string on failure, as opposed to just the input string.

Reply:
I've changed them to just reporting the encoded string in case of failure; that should still be sufficient to notice the offending test case, and keeps working if we'd add cases that have non-printable input. Ok?